Templates/PythonToolGem/Template/Code/Source/${Name}ModuleInterface.cpp

```diff
@@ -0,0 +1,32 @@
+// {BEGIN_LICENSE}
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+// {END_LICENSE}
+
+#include "${Name}ModuleInterface.h"
+#include <AzCore/Memory/Memory.h>
+
+#include <${Name}/${Name}TypeIds.h>
+
+namespace ${SanitizedCppName}
+{
+    AZ_TYPE_INFO_WITH_NAME_IMPL(${SanitizedCppName}ModuleInterface,
+        "${SanitizedCppName}ModuleInterface", ${SanitizedCppName}ModuleInterfaceTypeId);
+    AZ_RTTI_NO_TYPE_INFO_IMPL(${SanitizedCppName}ModuleInterface, AZ::Module);
+    AZ_CLASS_ALLOCATOR_IMPL(${SanitizedCppName}ModuleInterface, AZ::SystemAllocator);
+
+    ${SanitizedCppName}ModuleInterface::${SanitizedCppName}ModuleInterface()
+    {
+    }
+
+    AZ::ComponentTypeList ${SanitizedCppName}ModuleInterface::GetRequiredSystemComponents() const
+    {
+        return AZ::ComponentTypeList{
```

Comment:
Might be nice to have a comment here stating `This template has no c++ system component, but if you create one, add the type here so it is automatically activated`

Reply:
This does have a System Component.
It is added in the derived [EditorModule](https://github.com/o3de/o3de/blob/36b98f01825e0bffdbf343d2d54eb36f64c2ff92/Templates/PythonToolGem/Template/Code/Source/Tools/%24%7BName%7DEditorModule.cpp#L47-L56).